### DISCUSSION TITLE
output flag seems to ignore rules

### DISCUSSION THREAD

**Discussion by Southgarden116** [remarkjs/remark#974] [Q&A] [answered]
Hey, first of all thanks for all your great work here!

In my `.remarkrc` file I added some recommended preset rules, which show the warnings by running `npm run format` just fine.
However, adding the flag `--output`, changed things that were not showing before. I then noticed that there are optional settings with `remark-stringify` that I could use, which definitely helped.
What is still bothering me though is the inability to skip certain rules. I disabled a rule with `["remark-lint-no-undefined-references", false]` to not warn about shortcut references e.g. `[name]`, since we don't use them as references. But when I run `npm run format --output`, the first bracket is still being escaped instead of being ignored. Is there a way I could configure this?

**Comment by wooorm** [comment, marked as answer]
Linting and formatting are separate steps. See https://github.com/remarkjs/remark-lint#example-check-and-format-markdown-on-the-cli and https://github.com/remarkjs/remark-lint/pull/279.

There is no way to *not* escape references. It is intentional that opening brackets in text are escaped.

**Reply by Southgarden116** [reply]
Thanks for the reply!

**Comment by subfuzion** [comment]
Unfortunately that breaks checkboxes for those of us who like remark formatting, but would like to support extended syntax exceptions. Any recommendations?

**Reply by subfuzion** [reply]
Ah, I got this to work for my use case:

  1. Use remark-gfm to parse GitHub Flavored Markdown including
  task lists
  2. Disable no-undefined-references to prevent false positives

Maybe this will help someone else.
